app/forms/decidim/vocdoni/admin/census_permissions_form.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Vocdoni
+    module Admin
+      class CensusPermissionsForm < Decidim::Form
+        mimic :census_permissions
+
+        attribute :census_permissions, Array[String]
+
+        def data
+          verification_types = census_permissions
+
+          verified_users_ids = Decidim::Authorization
+                               .where(name: verification_types)
+                               .group(:decidim_user_id)
+                               .having("COUNT(distinct name) = ?", verification_types.count)
+                               .pluck(:decidim_user_id)
+
+          Decidim::User.where(id: verified_users_ids).uniq
```

Comment:
```suggestion

          users = context.current_organization.users.not_deleted.confirmed

          unless verification_types.blank?
            verified_users = Decidim::Authorization.select(:decidim_user_id)
                             .where(name: verification_types)
                             .group(:decidim_user_id)
                             .having("COUNT(distinct name) = ?", verification_types.count)

            users = users.where(id: verified_users)
          end
          
          users
```

I think we can optimize this query and ensure that only users from our organization are listed.
On the other hand this also returns the whole user database if no permissions are selected